### PR TITLE
Moving configuration values around

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

The PR moving `network_config.enable_http_monitoring` to `service_monitoring_config.enable_http_monitoring` while keeping backward compatibility and allowing customers enabling http monitoring by both options.

### Motivation

Organizing the configuration files without breaking customer's usage

### Additional Notes
viper PR - https://github.com/DataDog/viper/pull/27

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -572,6 +572,8 @@ require (
 
 replace github.com/pahanini/go-grpc-bidirectional-streaming-example v0.0.0-20211027164128-cc6111af44be => github.com/DataDog/go-grpc-bidirectional-streaming-example v0.0.0-20221024060302-b9cf785c02fe
 
+replace github.com/DataDog/viper v1.12.0 => github.com/guyarb/viper v0.0.0-20230225205112-9debdabead64
```
Comment (guyarb) [inline go.mod:575]: should be changed once https://github.com/DataDog/viper/pull/27 is merged

File: pkg/config/system_probe.go [hunk 2]
```diff
@@ -164,16 +164,21 @@ func InitSystemProbeConfig(cfg Config) {
...
 	cfg.BindEnvAndSetDefault(join(spNS, "dest_excludes"), map[string][]string{})
 
-	// network_config namespace only
-	cfg.BindEnv(join(netNS, "enable_http_monitoring"), "DD_SYSTEM_PROBE_NETWORK_ENABLE_HTTP_MONITORING")
-	cfg.BindEnv(join(netNS, "enable_https_monitoring"), "DD_SYSTEM_PROBE_NETWORK_ENABLE_HTTPS_MONITORING")
+	// service monitoring namespace only
+	cfg.BindEnv(join(smNS, "enable_http_monitoring"))
+	// Setting join(smNS, "enable_http_monitoring") as alias to join(netNS, "enable_http_monitoring"), will allow us to
+	// either set the field in the configuration file in either of the options.
+	cfg.SetAliasAndEnv(join(netNS, "enable_http_monitoring"), join(smNS, "enable_http_monitoring"), "DD_SYSTEM_PROBE_NETWORK_ENABLE_HTTP_MONITORING")
```
Comment (val06) [inline pkg/config/system_probe.go:171]: do we want to keep the env variable with `NETWORK` inside of it or should we introduce a new variable as well and keep the old one for backward compatibility?

**Review by val06 — COMMENTED** [review]
added a comment

File: pkg/config/system_probe.go (hunk 2 above)
Comment (guyarb) [reply pkg/config/system_probe.go:171]: I'm keeping it for backward compatibility, we have the new way of `DD_SERVICE_MONITORING_CONFIG_ENABLE_HTTP_MONITORING`

**Review by val06 — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Pull request closed without merging**